server/events/vcs/gitlab_client.go

```diff
@@ -198,14 +198,23 @@ func (g *GitlabClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 		return false, err
 	}
 
+	// Prevent nil pointer error when mr.HeadPipeline is empty
```

Comment:
Please add a test for this

Reply:
I just haven't got the time yet. I'm gonna try doing this over the long weekend here and get back to you

Reply:
@GenPage finally got the time to work on this, the tests are here, could you review them?